packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/bedrock/invoke-model.ts

```diff
@@ -102,9 +132,23 @@ export interface BedrockInvokeModelProps extends sfn.TaskStateBaseProps {
    * If you specify this field, the API response body is replaced with a reference to the
    * output location.
    *
-   * @default The API response body is returned in the result.
+   * @default - The API response body is returned in the result.
    */
   readonly output?: BedrockInvokeModelOutputProps;
+
+  /**
+   * The guardrail is applied to the invocation
+   *
+   * @default - No guardrail is applied to the invocation.
+   */
+  readonly guardrailConfiguration?: GuardrailConfiguration;
```

Comment:
```suggestion
  /**
   * The unique identifier of the guardrail that you want to use.
   * If you don't provide a value, no guardrail is applied to the invocation.
   */
  readonly guardrailIdentifier?: string;

  /**
   * The version number for the guardrail.
   * The value can also be `DRAFT`.
   */
  readonly guardrailVersion?: string;
```
To provide a simpler interface, I'd rather expose the two attributes directly and apply the `guardrailVersion must be specified if guardrailIdentifier` is specified validation.
What do you think?
Also, we should apply validation for the accepted values according to the [docs](https://docs.aws.amazon.com/bedrock/latest/APIReference/API_runtime_InvokeModel.html#API_runtime_InvokeModel_RequestSyntax)

Reply:
@lpizzinidev 
Thank you. 

Since guardrailIdentifier and guardrailVersion need to be used as a set, I thought it would be better to have them as required fields in the current standalone interface. What do you think?

For your reference, there was a similar discussion [here](https://github.com/aws/aws-cdk/pull/30319#discussion_r1612489270).

I also like the flat approach, but it could lead to an increase in validation patterns, and since bedrock is a growing service, there's a possibility of more guardrail fields being added in the future.

Adding accepted values validation is agree.

Reply:
I remember having similar discussions in the past in favor of the plain interface approach for other scenarios (as suggested by the [design guidelines](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#flat)).
Keeping the nested structure saves us from implementing explicit validation.
However, providing a clean interface for other programming languages is also a concern.

> since bedrock is a growing service, there's a possibility of more guardrail fields being added in the future

That's a good point. Let's keep it like this and see what the core team thinks about it 👍

Reply:
Thank you. For now, we've decided to keep the property design approach as is, without any modifications.

I'd like to wait for the maintainer's review.

I have added the validation.​​​​​​​​​​​​​​​​

Reply:
Hi both, the teams current stance is that we would prefer to have this interface containing the two required properties rather than exposing two separate attributes as it allows us to avoid adding more error handling and just enforce that both attributes are specified when defining the resource. 

While the section linked from the Design Guidelines does state that flat property structure is preferred over a nested structure, that is only the case when the two properties are not required to be specified together. I can update that to clarify the distinction.

Reply:
Both sides are technically correct here. In this case, I'd suggest using `guardrail` and instead of a nested set of values, make it a class with a single function so that it would look something like `guardrail: Guardrail.apply('my-guardrail', 1)`, even better, make the second value optional and if it isn't supplied, we automatically set that value to `DRAFT` for the user.

But also, if you can think of better naming conventions than mine, that would probably be good. I'm terrible at naming anything.

Reply:
@TheRealAmazonKendra  @paulhcsun 
Thank you. I think that approach is good too.
However, in this case, there's a possibility of unintentionally applying the DRAFT version of the guardrail. 
Since guardrail is a security-related configuration, I thought it might be better to make the version a required field in the current implementation, as it is now. This would help prevent the application of unintended versions. 
What do you think about this?

Also, I've modified the property name to `guardrail` as I thought it would be better.

Reply:
Totally fair concern, in that case, I would create an enum like class here for `guardrail`

```
public class Guardrail {
  public static enable(string: id, version: number);

  public static enableDraft(string: id);
}
```

Reply:
Thank you. I apologize for the repeated requests, but I still can't agree that the proposed method is better.

The guardrail version is a string in the L1 Construct, so using a method to make it a number would require casting.
Also, there's [validation for 'contentType'](https://github.com/aws/aws-cdk/pull/30426/files#diff-a9f9ef1a1d3f4c941af934cf7943b7be05c856d20246c0674fdfa72dacceb15fR284) which is not included in the guardrail, and [generation of policy documents](https://github.com/aws/aws-cdk/pull/30426/files#diff-a9f9ef1a1d3f4c941af934cf7943b7be05c856d20246c0674fdfa72dacceb15fR225). Because of these factors, I think the current approach of not separating classes is more understandable.
The proposed approach seems over-optimized to me.

If you still think the proposed method is better despite these points, could you provide an example of where a similar approach is used elsewhere? 
This would help me understand the implementation concept better.

Reply:
Hi @mazyu36, one example where a similar approach is used is in the [`Schedule` class in `aws-events`](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-events/lib/schedule.ts#L11-L70).

I think that casting from number to string is not a major concern here, and avoids us having to validate the version if a string is passed in. 

As for generating the policy documents, I think those can be left where it is, but I agree there's not a super clean way to validate for `contentType` without passing in the `contentType` to the method which is not a straightforward DX. I will discuss with the team and get back to you on that tomorrow.

Reply:
Hi @mazyu36, I believe the only allowed value we currently accept for `contentType` is `application/json`. I assume that when this module was first written it was made to be an optional property to reserve the right to support different `contentType`s in the future but for the time being I think that we should just deprecate the property and remove this check. If the day comes that we need to support additional `contentType`s then we can just undeprecate it.

Reply:
@paulhcsun 
Thank you. I've added the `GuardRail` Class and made `contentType` deprecated.